output/schema/schema.json

```diff
@@ -25416,7 +25416,7 @@
         {
           "description": "The query vector",
           "name": "query_vector",
-          "required": true,
+          "required": false,
```

Comment:
Is there a way to enforce that one of query_vector or query_vector_builder are defined through TS and/or the spec? The way the type would work if we were to merge this would inform the user that both fields are optional

Reply:
Not that I know of. I just copied this over from the `KnnQuery` Interface, because I saw it was out of date and I thought I might as well fix that clearly out of date part of the code. 

I could just remove this part from the MR and this can be fixed another time. The Java Client only needs the `KnnQuery` interface part, as far as I can tell. That way the similarity field would work and people could use it. Should I do that?

Reply:
@mshameti 

> Is there a way to enforce that one of query_vector or query_vector_builder are defined through TS and/or the spec?

We can do this:
```ts
export type KnnQuery = {
  /** The name of the vector field to search against */
  field: Field
  /** The final number of nearest neighbors to return as top hits */
  k: integer
  /** The number of nearest neighbor candidates to consider per shard */
  num_candidates: integer
  /** The minimum similarity required for a document to be considered a match */
  similarity?: float
  /** Boost value to apply to kNN scores */
  boost?: float
  /** Filters for the kNN search query */
  filter?: QueryContainer | QueryContainer[]
} & (
  {
    /** The query vector */
    query_vector: QueryVector
  } & {
    /** The query vector builder. You must provide a query_vector_builder or query_vector, but not both. */
    query_vector_builder: QueryVectorBuilder
  }
)
```

But we will get this error:
```
Error: Unhandled node kind: IntersectionType
```